rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -109,43 +58,94 @@ impl MultisigCheckpointSyncer {
             return Ok(None);
         }
 
-        // Sort in descending order to iterate through higher indices first.
+        // Sort in descending order. The n'th index will represent
+        // the highest index for which we (supposedly) have (n+1) signed checkpoints
         latest_indices.sort_by(|a, b| b.cmp(a));
-
-        let mut last_processed_index = 0;
-
-        // Try to find a quorum among the latest indices
-        let mut index_count = 0;
-        for latest_index in &latest_indices {
-            if *latest_index != last_processed_index {
-                last_processed_index = *latest_index;
-                index_count = 1;
-            } else {
-                index_count += 1;
-            }
-
-            // If we've found a quorum, return it
-            if index_count >= self.threshold {
-                return Ok(Some(last_processed_index));
+        if let Some(highest_quorum_index) = latest_indices.get(threshold - 1) {
+            let mut i = std::cmp::min(*highest_quorum_index, maximum_index);
```

Comment:
nit: took me a sec to understand what `i` was intended for, a non-single-letter name may be nice